Repository/RevisionRepository.php

```diff
@@ -576,57 +576,86 @@ public function deleteRevisions(ContentType $contentType = null)
         }
     }
 
-    /**
-     * @param ContentType $contentType
-     * @param \DateTime $until
-     * @param string $by
-     * @param bool $force
-     *
-     * @param bool $id
-     * @return int affected rows
-     * @throws \Exception
-     */
-    public function lockRevisions(ContentType $contentType, \DateTime $until, $by, $force = false, $id = false)
+    public function lockRevisions(?ContentType $contentType, \DateTime $until, $by, $force = false, $id = false): int
     {
-        $params = ['by' => $by, 'until' => $until, 'content_type' => $contentType];
-
-        /** @var QueryBuilder $qbSelect */
         $qbSelect = $this->createQueryBuilder('s');
         $qbSelect
             ->select('s.id')
-            ->andWhere($qbSelect->expr()->eq('s.contentType', ':content_type'))
             ->andWhere($qbSelect->expr()->isNull('s.endTime'))
             ->andWhere($qbSelect->expr()->eq('s.deleted', $qbSelect->expr()->literal(false)))
             ->andWhere($qbSelect->expr()->eq('s.draft', $qbSelect->expr()->literal(false)))
         ;
 
+        $qbUpdate = $this->createQueryBuilder('r');
+        $qbUpdate
+            ->update()
+            ->set('r.lockBy', ':by')
+            ->set('r.lockUntil', ':until')
+            ->setParameters(['by' => $by, 'until' => $until]);
+
+        if (null !== $contentType) {
+            $qbSelect->andWhere($qbSelect->expr()->eq('s.contentType', ':content_type'));
+            $qbUpdate->setParameters(['content_type' => $contentType]);
+        }
+
         if (!$force) {
             $qbSelect->andWhere($qbSelect->expr()->orX(
                 $qbSelect->expr()->lt('s.lockUntil', ':now'),
                 $qbSelect->expr()->isNull('s.lockUntil')
             ));
-
-            $params['now'] = new \DateTime();
+            $qbUpdate->setParameters(['now' => new \DateTime()]);
```

Comment:
```suggestion
            $qbUpdate->setParameters('now', new \DateTime());
```
Shoiuld be setParameter, setParmaters will not do a merge

Reply:
(thanks for that)